runtime/parser2/parser.go

```diff
@@ -45,6 +46,8 @@ type parser struct {
 	backtrackingCursorStack []int
 	// bufferedErrorsStack is the stack of parsing errors encountered during buffering
 	bufferedErrorsStack [][]error
+	// memoryGauge is used to meter memory usage
+	memoryGauge common.MemoryGauge
```

Comment:
This field seems unused

Reply:
Right, and the parser/lexer integration was not working properly at all. Fixed this in cbe726b01ea028157cc902e342ea2cd59fce4837